epochStart/metachain/economics.go

```diff
@@ -406,6 +419,54 @@ func (e *economics) startNoncePerShardFromLastCrossNotarized(metaNonce uint64, e
 	return mapShardIdNonce, nil
 }
 
+func (e *economics) checkEconomicsInvariants(
+	computedEconomics block.Economics,
+	inflationRate float64,
+	maxBlocksInEpoch uint64,
+	totalNumBlocksInEpoch uint64,
+	metaBlock *block.MetaBlock,
+	epoch uint32,
+) error {
+	if epoch <= e.stakingV2EnableEpoch {
+		return nil
+	}
+
+	if !core.IsInRangeInclusiveFloat64(inflationRate, 0, e.rewardsHandler.MaxInflationRate(1)) {
+		return epochStart.ErrInvalidInflationRate
```

Comment:
suggestion to wrap all these errors with the compared value & limits. It will be easier to debug.

Reply:
done